src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -415,17 +418,21 @@ def unique(self, level: Hashable | None = None) -> Index:
         See Also
         --------
         unique : Numpy array of unique values in that column.
-        Series.unique : Return unique values of Series object.
+        Series.unique : Return unique values of a Series object.
 
         Examples
         --------
         >>> idx = pd.Index([1, 1, 2, 3, 3])
         >>> idx.unique()
         Index([1, 2, 3], dtype='int64')
         """
-        # TODO: SNOW-1458132 implement unique
-        WarningMessage.index_to_pandas_warning("unique")
-        return Index(self.to_pandas().unique(level=level))
+        # TODO: SNOW-1514782 MultiIndex - support level for Index.unique.
+        if level is not None:
+            ErrorMessage.not_implemented(
```

Comment:
I think our coverage check will require you to add a test for `level`.

Reply:
I got rid of the level check here since Yun mentioned that this logic is already taken care of [here](https://github.com/snowflakedb/snowpark-python/pull/1875/files/2f2f924670b5f9fbd446c1825c83a43619e19c8a#r1669002320).